crates/environ/src/module_artifacts.rs

```diff
@@ -275,6 +275,15 @@ impl<'a> ObjectBuilder<'a> {
             self.push_debug(&mut dwarf, &debuginfo.dwarf.debug_str_offsets);
             self.push_debug(&mut dwarf, &debuginfo.debug_ranges);
             self.push_debug(&mut dwarf, &debuginfo.debug_rnglists);
+
+            if let Some(dwarf_package) = debuginfo.dwarf_package {
+                self.push_debug(&mut dwarf, &dwarf_package.debug_abbrev);
+                self.push_debug(&mut dwarf, &dwarf_package.debug_info);
+                self.push_debug(&mut dwarf, &dwarf_package.debug_line);
+                self.push_debug(&mut dwarf, &dwarf_package.debug_str);
+                self.push_debug(&mut dwarf, &dwarf_package.debug_str_offsets);
+                self.push_debug(&mut dwarf, &dwarf_package.debug_rnglists);
+            }
```

Comment:
If these sections are present in the dwarf package are they not present in the original wasm? If so could that be validated here because otherwise this might create dual sections which might confuse some dwarf readers.

Reply:
From Appendix F in the spec https://dwarfstd.org/doc/DWARF5.pdf:

![image](https://github.com/bytecodealliance/wasmtime/assets/2720041/e86a7273-6cd3-4a39-a549-b2f866b2a7f6)


So they are there, optionally, but with a `.dwo` suffix.  Perhaps I don't need to push any here, which would solve things.

Reply:
Ah ok, sounds good! You could also update `push_debug` to take an optional suffix perhaps of what to use as the new section name and set that to `".dwo"` for these method calls

Reply:
I didn't need these after all, so have removed this change.